### PR TITLE
Some fixes in preparation for reading types from Scala 2 pickles

### PR DESCRIPTION
**Pull request by sjrd**
The test, even in its truncated form, needs the first two commits of this PR to pass.

### CODE REVIEW

File: shared/src/main/scala/tastyquery/Contexts.scala [hunk 1]
```diff
@@ -188,14 +188,10 @@ object Contexts {
 
     def moduleRoot: RegularSymbol = Contexts.moduleRoot(classRoot)(using this)
 
-    def createSymbol[T <: Symbol](name: Name, factory: SymbolFactory[T], addToDecls: Boolean = false): T =
-      owner.getDeclInternal(name) match
```
Comment (bishabosha) [inline shared/src/main/scala/tastyquery/Contexts.scala:192]: is there a reason to get rid of this guard? I guess its for overloads

Comment (sjrd) [reply shared/src/main/scala/tastyquery/Contexts.scala:192]: Yes, it's for overloads.

**Comment by bishabosha** [comment]
is this PR still relevant?

**Comment by sjrd** [comment]
Superseded by #63.

**Pull request closed without merging**